src/containerapp/azext_containerapp/dotnet_component_decorator.py

```diff
@@ -0,0 +1,81 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from typing import Any, Dict
+
+from azure.cli.core.commands import AzCliCommand
+from azure.cli.core.azclierror import CLIInternalError
+from azure.cli.command_modules.containerapp.base_resource import BaseResource
+from knack.log import get_logger
+
+from ._models import DotNetComponent as DotNetComponentModel
+
+from ._client_factory import handle_raw_exception
+from ._constants import DOTNET_COMPONENT_RESOURCE_TYPE
+
+logger = get_logger(__name__)
+
+
+class DotNetComponentDecorator(BaseResource):
+    def __init__(self, cmd: AzCliCommand, client: Any, raw_parameters: Dict, models: str):
+        super().__init__(cmd, client, raw_parameters, models)
+        self.dotnet_component_def = DotNetComponentModel
+
+    def get_argument_environment_name(self):
+        return self.get_param("environment_name")
+
+    def get_argument_dotnet_component_name(self):
+        return self.get_param("dotnet_component_name")
+
+    def get_argument_component_type(self):
+        return self.get_param("dotnet_component_type")
+
+    def set_argument_component_type(self, component_type: str):
+        self.set_param("dotnet_component_type", component_type)
+
+    def construct_payload(self):
+        if self.get_argument_component_type() == "":
```

Comment:
If you using `arg_type=get_enum_type(['AspireDashboard'])` in the params, not need this check.

Reply:
The expected behavior is we need to default the component type to `AspireDashboard` if customer did not provide a `--type` argument. so this check is still required?

Reply:
Keeping this as-is because of [this response](https://github.com/Azure/azure-cli-extensions/pull/7564#discussion_r1597921969)